src/main/java/org/jnode/fs/xfs/Superblock.java

```diff
@@ -58,9 +58,21 @@ public Superblock(XfsFileSystem fileSystem) throws FileSystemException {
      *
      * @return the block size.
      */
-    public int getBlockSize() {
-        return (int) getUInt32(0x4);
-    }
+    public long getBlockSize() { return getUInt32(0x4); }
```

Comment:
```suggestion
    public long getBlockSize() { 
        return getUInt32(0x4); 
    }
```
This and the following functions.

Reply:
The code changes were applied. thanks.